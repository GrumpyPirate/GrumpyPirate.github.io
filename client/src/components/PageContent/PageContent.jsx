import React from 'react';
import PropTypes from 'prop-types';

import './PageContent.scss';

const PageContent = ({ children }) => (
  <div className="page-content">{children}</div>
);

PageContent.propTypes = {
  children: PropTypes.node.isRequired,
};

export default PageContent;
