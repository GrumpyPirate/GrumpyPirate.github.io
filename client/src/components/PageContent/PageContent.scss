/* ---------------------------------------------------------------------------------------------------------------------
** Global layout styles that apply to all document content
** ------------------------------------------------------------------------------------------------------------------ */
@import 'functions';
@import 'variables';
@import 'mixins';

$page-padding: 3rem;

.page-content {
  text-align: center;
  padding-top: ($page-padding / 2);
	padding-bottom: ($page-padding / 2);

	// md up
	@include respond-up(md) {
    padding-top: $page-padding;
		padding-bottom: $page-padding;
	}
}
