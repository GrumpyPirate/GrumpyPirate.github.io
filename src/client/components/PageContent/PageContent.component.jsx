import React from 'react';
import PropTypes from 'prop-types';

import classes from './PageContent.component.scss';

const PageContent = ({ children }) => <div className={classes['page-content']}>{children}</div>;

PageContent.propTypes = {
  children: PropTypes.node.isRequired,
};

export default PageContent;
