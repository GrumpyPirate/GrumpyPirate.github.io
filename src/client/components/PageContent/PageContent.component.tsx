import React, { FunctionComponent } from 'react';

import { ChildrenProps } from 'types';

import classes from './PageContent.component.scss';

const PageContent: FunctionComponent<ChildrenProps> = ({ children }: ChildrenProps) => (
  <div className={classes['page-content']}>{children}</div>
);

export default PageContent;
