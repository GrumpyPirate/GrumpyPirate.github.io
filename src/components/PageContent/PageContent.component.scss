@import 'core';

$page-padding: 3rem;

.page-content {
  padding-bottom: ($page-padding / 2);
  padding-top: ($page-padding / 2);
  text-align: center;

  @include respond-up('md') {
    padding-bottom: $page-padding;
    padding-top: $page-padding;
  }
}
