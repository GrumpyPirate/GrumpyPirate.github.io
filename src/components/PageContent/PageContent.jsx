// React
import React, { Component } from 'react'

// Component
class PageContent extends Component {
  constructor(props) {
    super(props)
  } // /constructor(props)

  render () {
    const alignClass = this.props.align
      ? ` text-${this.props.align}`
      : ''

    return (
      <div className={`page__content${alignClass}`}>
        <div className="container-fluid">
          <div className="row justify-content-center">
            <div className="col-12 col-sm-10 col-lg-12 col-xl-10">
              {this.props.children}
            </div>
          </div>
        </div>
      </div>
    )
  } // /render ()
} // /class PageContent extends Component

export default PageContent
