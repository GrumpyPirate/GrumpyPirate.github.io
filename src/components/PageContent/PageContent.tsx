import React, { FunctionComponent } from 'react';

import { ChildrenProps } from 'types/common';

import classes from './PageContent.scss';

const PageContent: FunctionComponent<ChildrenProps> = ({ children }) => (
  <div className={classes['page-content']}>{children}</div>
);

export default PageContent;
