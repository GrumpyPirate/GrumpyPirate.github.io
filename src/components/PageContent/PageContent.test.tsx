import { shallow } from 'enzyme';
import React from 'react';

import PageContent from './PageContent';
import fixture from './PageContent.fixture';

describe('Components', () => {
  describe('PageContent', () => {
    const props = {
      ...fixture,
    };

    it('should render as expected, without crashing', () => {
      const wrapper = shallow(<PageContent {...props} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
