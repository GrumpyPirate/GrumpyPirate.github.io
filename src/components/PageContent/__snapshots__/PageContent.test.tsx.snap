// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components PageContent should render as expected, without crashing 1`] = `
.c0 {
  position: relative;
  padding-bottom: 1.5rem;
  padding-top: 1.5rem;
  text-align: center;
  min-height: 100vh;
}

@media (min-width:48em) {
  .c0 {
    padding-bottom: 3rem;
    padding-top: 3rem;
  }
}

<div
  class="c0"
>
  <div>
    Test
  </div>
</div>
`;
